experimental/origin-dapp2/src/pages/growth/Campaigns.js

```diff
@@ -0,0 +1,615 @@
+  const { campaign, selected, onClick } = props
+  const completedIndicator =
+    campaign.status === 'completed' || campaign.status === 'capReached'
+
+  let statusClass = ''
+  if (campaign.status === 'active') {
+    statusClass = 'active'
+  } else if (campaign.status === 'pending') {
+    statusClass = 'inactive'
+  }
+
+  return (
+    <a
+      href="#"
+      onClick={e => {
+        e.preventDefault()
+        onClick(campaign.id)
+      }}
+      className="pt-4 pb-4 pl-3 pr-3"
+    >
+      <div className="campaign d-flex flex-column align-items-center">
+        <div className={`status ${statusClass}`}>
+          {completedIndicator && <img src="images/circular-check-button.svg" />}
+        </div>
+        <div className={`name ${selected ? 'active' : ''}`}>
+          {campaign.name}
+        </div>
+        {selected && <div className="select-bar" />}
+      </div>
+    </a>
+  )
+}
+
+function CampaignNavList(props) {
+  const { campaigns, onCampaignClick, selectedCampaignId } = props
+  return (
+    <div className="campaign-list d-flex justify-content-center mt-4">
+      {campaigns.map(campaign => (
+        <CampaignNavItem
+          key={campaign.id}
+          campaign={campaign}
+          selected={campaign.id === selectedCampaignId}
+          onClick={onCampaignClick}
+        />
+      ))}
+    </div>
+  )
+}
+
+class ProgressBar extends Component {
+  constructor(props) {
+    super(props)
+    this.triggerAnimation = true
+  }
+
+  render() {
+    const { progress } = this.props
+
+    /* For triggering animation first render of react component needs to set
+     * the width to 0. All subsequent renders set it to the actuall value.
+     */
+    const triggerAnimationThisFrame = this.triggerAnimation
+    if (progress > 0 && this.triggerAnimation) {
+      setTimeout(() => {
+        this.triggerAnimation = false
+        this.forceUpdate()
+      }, 250)
+    }
+
+    return (
+      <Fragment>
+        <div className="campaign-progress mt-3">
+          <div className="background" />
+          {progress > 0 && (
+            <div
+              className="foreground"
+              style={{
+                width: `${!triggerAnimationThisFrame ? progress : '0'}%`
+              }}
+            />
+          )}
+        </div>
+        <div className="indicators d-flex justify-content-between mt-2">
+          <div>0</div>
+          <div>25</div>
+          <div>50</div>
+          <div>75</div>
+          <div>100</div>
+        </div>
+      </Fragment>
+    )
+  }
+}
+
+function Action(props) {
+  let { type, status, reward, rewardEarned, rewardPending } = props.action
+  const showLockIcon = status === 'inactive'
+
+  const actionCompleted = ['exhausted', 'completed'].includes(status)
+  let backgroundImgSrc = actionCompleted
+    ? 'images/identity/verification-shape-green.svg'
+    : 'images/identity/verification-shape-blue.svg'
+
+  const formatTokens = tokenAmount => {
+    return web3.utils
```

Comment:
We should try to avoid any web3 calls in dapp2 and rely on this coming from GraphQL instead

Reply:
Got the idea from here: https://github.com/OriginProtocol/origin/blob/master/experimental/origin-dapp2/src/hoc/withTokenBalance.js#L26

😈 

Will resolve by adding bignumber.js npm package.